### PR TITLE
Bounce address

### PR DESCRIPTION
**Pull request by kpmeen**
Added support for setting a custom bounce address as specified in commons mail (here: http://commons.apache.org/proper/commons-email/userguide.html#Handling_Bounced_Messages).

### CODE REVIEW

**Comment by Mogztter** [comment]
Hi @kpmeen,

Thank you for your contribution.  Would you mind signing our CLA?

http://www.typesafe.com/contribute/cla

Thanks!

File: src/main/scala/play/api/libs/mailer/MailerPlugin.scala [hunk 1]
```diff
@@ -182,6 +185,7 @@ abstract class CommonsMailer(smtpHost: String, smtpPort: Int,
     email.setSubject(data.subject)
     email.setFrom(data.from)
     data.replyTo.foreach(setAddress(_) { (address, name) => email.addReplyTo(address, name)})
+    data.bounceAddress.map(email.setBounceAddress)
```
Comment (Mogztter) [inline src/main/scala/play/api/libs/mailer/MailerPlugin.scala]: Can you use `data.bounceAddress.foreach(email.setBounceAddress)` to stay consistent with the rest of the code ?

**Comment by Mogztter** [comment]
Could you do a rebase to avoid the merge commit 426a734a46d2d52d6c4c292e348522ec40b517eb ?

**Comment by kpmeen** [comment]
Thanks for swift feedback! Updated and pushed :-)

**Comment by jroper** [comment]
lgtm

**Comment by Mogztter** [comment]
Yep :+1:

**Comment by Mogztter** [comment]
Great work, thanks again!

**Comment by mkurz** [comment]
@Mogztter Could it be that for [adding the bounce address](https://github.com/playframework/play-mailer/blob/master/src/main/scala/play/api/libs/mailer/MailerPlugin.scala#L134) we also could run into encoding problems like #28? Shouldn't `setAddress` also get called here? I am not sure, I was just thinking about it during reviewing the code...

**Comment by Mogztter** [comment]
AFAIK there's only one method to set the bounce address:
https://commons.apache.org/proper/commons-email/apidocs/org/apache/commons/mail/Email.html#setBounceAddress(java.lang.String)

This means that you are not supposed to define a name (just an email
address).
If we define a name and an email I think we will have encoding problems...
We need to add a test to verify that assertion and open an issue in commons
mailer ;)

**Pull request merged**